ethcore/spec/src/spec.rs

```diff
@@ -131,6 +131,7 @@ fn run_constructors<T: Backend>(
 	let start_nonce = engine.account_start_nonce(0);
 
 	let mut state = State::from_existing(db, root, start_nonce, factories.clone())?;
+	state.populate_from(genesis_state.clone());
 
 	// Execute contract constructors.
 	let env_info = EnvInfo {
```

Comment:
So this is not used when `self.constructors` is empty, nor is `from` below. Maybe wrap the whole block in an `if !constructors.empty()` and move the `state.commit()` outside?

Reply:
I don't have a strong opinion about this change, but it's out of scope of this PR, feel free to commit it directly to your branch after this PR is merged.